packages/local_auth/lib/local_auth.dart

```diff
@@ -101,6 +101,8 @@ class LocalAuthentication {
     bool biometricOnly = false,
   }) async {
     assert(localizedReason != null);
+    assert(localizedReason.isNotEmpty);
```

Comment:
What's the behavior on android if this is empty?

Reply:
On Android 11 (simulator) there are no problems if the `localizedReason` is empty. Is it a good idea to add the assert only on iOS case using `Platform.isIOS`?

Reply:
> Is it a good idea to add the assert only on iOS case using Platform.isIOS? 

Yes. If android can handle null, especially if null and empty string means different things on android, we should allow null on android.  Do you know if there are difference between null and empty string on android?

Reply:
Looking at this again. I think you are right, we really shouldn't allow empty strings or null on android either. So the change looks good.

Reply:
@enricobenedos Let's make sure to mention the `localizedReason` must not be an empty String in the documentation.

Reply:
Hello @cyanglaz, sorry for the delay but I'm a little bit busy. I'm happy that you check that is it a good idea to also not give the possibility to input an empty string on Android. I will update the docs soon.

Reply:
Awesome! Thanks, once you update the doc we will land this!

Reply:
I've updated the `authenticate` method documentation.